services/dbService.tsx

```diff
@@ -1,53 +1,64 @@
-import { db } from '../firebase/firebase.config';
-import { collection, addDoc, getDocs, QuerySnapshot, DocumentData } from "firebase/firestore"; 
-import { GeoPoint } from 'firebase/firestore';
+import {db} from '../firebase/firebase.config';
+import {collection, addDoc, getDocs, getDoc, doc} from "firebase/firestore";
 import Event from '../models/event';
 
 
 export class EventDoc {
     id: string
     name: string
     description: string
+    latitude: number
+    longitude: number
+    location: any
 
-    constructor(id: string, name: string, description: string) {
-      this.id = id;
-      this.name = name;
-      this.description = description;
-  }
+    constructor(id: string, name: string, description: string, latitude: number, longitude: number, location: any) {
+        this.id = id;
+        this.name = name;
+        this.description = description;
+        this.latitude = latitude;
+        this.longitude = longitude;
+        this.location = location
+    }
 }
 
-export async function addEvent(event: Event){
+export async function addEvent(event: Event) {
     try {
         const docRef = await addDoc(collection(db, "events"), {
-        id: event.id,
-        name: event.name,
-        type: event.type,
-        description: event.description,
-        startDate: event.startDate,
-        endDate: event.endDate,
-        photo: event.photo,
-        postTime: event.postTime ,
-        stars: event.stars,
-        votes: event.votes
-      });
-      console.log("Document written with ID: ", docRef.id);
+            id: event.id,
+            name: event.name,
+            type: event.type,
+            description: event.description,
+            startDate: event.startDate,
+            endDate: event.endDate,
+            photo: event.photo,
+            postTime: event.postTime,
+            stars: event.stars,
+            votes: event.votes
+        });
+        console.log("Document written with ID: ", docRef.id);
     } catch (e) {
-      console.error("Error adding document: ", e);
+        console.error("Error adding document: ", e);
     }
-  }
+}
 
 
-export async function getEvents() : Promise<EventDoc[]> {
+export async function getEvents(): Promise<EventDoc[]> {
     const events = await getDocs(collection(db, "events"));
     let arr: EventDoc[] = [];
     if (events.size > 0) {
-      events.forEach(doc => {
-        arr.push(new EventDoc(doc.id, doc.data().name, doc.data().description))
-      })
+        events.forEach(doc => {
+            arr.push(new EventDoc(doc.id, doc.data().name, doc.data().description, doc.data().latitude, doc.data().longitude, doc.data().location))
+        })
+    } else {
+        // decide what you want to do if no results
```

Comment:
👀

Reply:
To akurat twoje :). Ja tylko formatowanie tutaj zmieniłem chyba.

Reply:
Jeżeli już to Adama